kore/src/Kore/Log.hs

```diff
@@ -127,6 +138,22 @@ withMainLogger
             . koreLogFilters koreLogOptions
             . makeKoreLogger exeName timestampsSwitch
 
+withMainReport
+    :: FilePath
+    -> KoreLogOptions
+    -> (LogAction IO ActualEntry -> IO a)
+    -> IO a
+withMainReport
+    reportDirectory
+    koreLogOptions@KoreLogOptions { timestampsSwitch, exeName }
+    continue
+  =
+    Colog.withLogTextFile ("./" <> reportDirectory <> "/LoggedErrors.txt")
+        $ continue
+        . koreLogTransformer koreLogOptions { logLevel = Error}
+        . koreLogFilters koreLogOptions { logLevel = Error}
```

Comment:
We should use the filters that the user applied. The user filters are applied in one place: `withMainLogger`, so this logger should be moved there.